### PR TITLE
Labels: Added plugins, adjusted labels

### PR DESCRIPTION
**Pull request by levb**

### CODE REVIEW

File: labels/mapping.go [hunk 1]
```diff
@@ -10,7 +36,7 @@ var core = []Label{
 	{"2: QA Review", "Requires review by a QA tester", "7cdfe2"},
 	{"3: Reviews Complete", "All reviewers have approved the pull request", "0e8a16"},
 	{"Awaiting Submitter Action", "Blocked on the author", "b60205"},
-	{"Do Not Merge/Awaiting PR", "Awaiting another pull request before merging (e.g. server changes)", "a32735"},
+	// {"Do Not Merge/Awaiting PR", "Awaiting another pull request before merging (e.g. server changes)", "a32735"},
```
Comment (levb) [inline labels/mapping.go]: arguably, no needed for plugins - Do Not Merge + a comment should be enough?

File: labels/mapping.go [hunk 2]
```diff
@@ -21,11 +47,11 @@ var core = []Label{
 
 var plugin = []Label{
 	{"Bug", "Something isn't working", "d73a4a"},
-	{"Duplicate", "This issue or pull request already exists", "cfd3d7"},
+	// {"Duplicate", "This issue or pull request already exists", "cfd3d7"},
```
Comment (levb) [inline labels/mapping.go]: I don't really see us using resolution statuses for plugins, closing the issue with a comment should be enough?

File: labels/mapping.go [hunk 3]
```diff
@@ -35,8 +61,10 @@ var helpWanted = []Label{
 	{"Good First Issue", "Suitable for first-time contributors", "7057ff"},
 	{"Hacktoberfest", "", "dc7d02"},
 	{"Help Wanted", "Community help wanted", "33aa3f"},
-	{"Needs spec", "Needs further specification to be a good help wanted ticket", "88acea"},
+	// {"Needs spec", "Needs further specification to be a good help wanted ticket", "88acea"},
```
Comment (levb) [inline labels/mapping.go]: We discussed that this may not be needed as a special label... Just a normal Help Wanted

File: labels/mapping.go [hunk 4]
```diff
@@ -1,7 +1,33 @@
...
+	"mattermost-plugin-gitlab",
+	"mattermost-plugin-jenkins",
+	"mattermost-plugin-jira",
+	"mattermost-plugin-mailermost",
+	"mattermost-plugin-suggestions",
+	"mattermost-plugin-webex",
+	"mattermost-plugin-welcomebot",
+	// "mattermost-plugin-workflow",
+	"mattermost-plugin-zoom",
+}
```
Comment (hanzei) [inline labels/mapping.go:24]: What was your guideline for picking these plugins? Why was e.g. mattermost-plugin-memes leaved out?

File: labels/mapping.go (hunk 1 above)
Comment (hanzei) [reply labels/mapping.go]: We use this label in the server and webapp a lot. 2/5 to have it here for consistency.

File: labels/mapping.go (hunk 2 above)
Comment (hanzei) [reply labels/mapping.go]: To me this helps making clear why an issue was closed. It's also a very common label on GH. 3/5

File: labels/mapping.go (hunk 3 above)
Comment (hanzei) [reply labels/mapping.go]: I agree. 0/5 if we need this.

File: labels/mapping.go [hunk 5]
```diff
@@ -21,11 +47,11 @@ var core = []Label{
 
 var plugin = []Label{
 	{"Bug", "Something isn't working", "d73a4a"},
-	{"Duplicate", "This issue or pull request already exists", "cfd3d7"},
+	// {"Duplicate", "This issue or pull request already exists", "cfd3d7"},
 	{"Enhancement", "New feature or request", "a2eeef"},
 	{"Needs Mattermost Changes", "Requires changes to the Mattermost Plugin tookit", "9c14c9"},
 	{"Question", "Further information is requested", "d876e3"},
-	{"Wontfix", "This will not be worked on", "ffffff"},
+	// {"Wontfix", "This will not be worked on", "ffffff"},
```
Comment (hanzei) [inline labels/mapping.go]: Same reason as above. I don't think a PR/issue could be closed without a label. This makes clear, why it got closed.

File: labels/mapping.go [hunk 6]
```diff
@@ -35,8 +61,10 @@ var helpWanted = []Label{
 	{"Good First Issue", "Suitable for first-time contributors", "7057ff"},
 	{"Hacktoberfest", "", "dc7d02"},
 	{"Help Wanted", "Community help wanted", "33aa3f"},
-	{"Needs spec", "Needs further specification to be a good help wanted ticket", "88acea"},
+	// {"Needs spec", "Needs further specification to be a good help wanted ticket", "88acea"},
 	{"Tech/Go", "", "0e8a16"},
+	{"Tech/JavaScript", "", "f9d0c4"},
+	{"Tech/TypeScript", "", "c9ffff"},
```
Comment (hanzei) [inline labels/mapping.go]: I'm would like to keep things simple and just have `Tech/ReactJS`. What is the value of distinguishing between the two types?

File: labels/mapping.go (hunk 3 above)
Comment (levb) [reply labels/mapping.go]: One can always create more labels in repos, and if this proves to be useful, we can add it to the list.

File: labels/mapping.go (hunk 4 above)
Comment (levb) [reply labels/mapping.go:24]: I just added the repos that I track/fully know about. Generally correlates with what I have been considering "production". I have no objection to extending the list, but also maybe 1/5 we should make the repo name an (optional) command flag? (defaults to the list)

File: labels/mapping.go (hunk 6 above)
Comment (levb) [reply labels/mapping.go]: TypeScript is in fashion...

File: labels/mapping.go (hunk 5 above)
Comment (levb) [reply labels/mapping.go]: could be closed without a label - not sure I agree, seems like it leads to more labels that we may not ever _query_ on. But, really 0/5. We can add them all for now, and then see if we actually use them.

File: labels/mapping.go (hunk 6 above)
Comment (hanzei) [reply labels/mapping.go]: /shrug-wick

Comment (hanzei) [reply labels/mapping.go]: /shrug_wic

Comment (hanzei) [reply labels/mapping.go]: Is this really a reason?

File: labels/mapping.go (hunk 4 above)
Comment (hanzei) [reply labels/mapping.go:24]: I like the idea :+1:. Let's go with this list (plus mattermost-plugin-workflow?) for now and iterate on it.

File: labels/mapping.go (hunk 6 above)
Comment (levb) [reply labels/mapping.go]: Yes. If you find it ridiculous, I will not disagree, and will remove the label.

Comment (hanzei) [reply labels/mapping.go]: Please remote it. Let's go with just the two "Tech"s for now.

File: labels/mapping.go [hunk 7]
```diff
@@ -1,42 +1,65 @@
...
-var plugin = []Label{
+var issue = []Label{
+	{"Triage", "", "efcb6e"},
 	{"Bug", "Something isn't working", "d73a4a"},
-	{"Duplicate", "This issue or pull request already exists", "cfd3d7"},
 	{"Enhancement", "New feature or request", "a2eeef"},
-	{"Needs Mattermost Changes", "Requires changes to the Mattermost Plugin tookit", "9c14c9"},
 	{"Question", "Further information is requested", "d876e3"},
+	{"Invalid", "This doesn't seem right", "e4e669"},
+	{"Duplicate", "This issue or pull request already exists", "cfd3d7"},
```
Comment (hanzei) [inline labels/mapping.go]: Would you please order the labels alphabetically?

File: labels/main.go [hunk 8]
```diff
@@ -12,12 +13,31 @@ import (
 
 const org = "mattermost"
 
+var useRepo = flag.String("repo", "", "Github Mattermost repository name")
+var useCoreLabels = flag.Bool("core-labels", true, "Core set of Mattermost labels")
```
Comment (hanzei) [inline labels/main.go]: Why would you use this labels if it's enabled by default and you can combine it with `usePluginLabels`?

I would like to be explicit we applying labels. What do you think about setting both flags to `false` by default and requiring at least one flag?

File: labels/mapping.go [hunk 9]
```diff
@@ -1,42 +1,65 @@
...
 var helpWanted = []Label{
 	{"Difficulty/1:Easy", "Easy ticket", "c2e0c6"},
 	{"Difficulty/2:Medium", "Medium ticket", "bfdadc"},
 	{"Difficulty/3:Hard", "Hard ticket", "f9d0c4"},
 	{"Good First Issue", "Suitable for first-time contributors", "7057ff"},
 	{"Hacktoberfest", "", "dc7d02"},
 	{"Help Wanted", "Community help wanted", "33aa3f"},
-	{"Needs spec", "Needs further specification to be a good help wanted ticket", "88acea"},
 	{"Tech/Go", "", "0e8a16"},
+	{"Tech/JavaScript", "", "f9d0c4"},
```
Comment (hanzei) [inline labels/mapping.go:62]: I don't see the value of this. This is just a super set of `Tech/ReactJS`, right?

File: labels/main.go (hunk 8 above)
Comment (hanzei) [reply labels/main.go]: Maybe we just remove this code part and add it in a separate PR?

File: labels/mapping.go [hunk 10]
```diff
@@ -1,42 +1,65 @@
 package main
 
-var mapping = map[string][]Label{
-	"mattermost-plugin-giphy": MergeLabels(core, plugin, helpWanted),
+var coreLabels = MergeLabels(pullRequest, issue, helpWanted, docs)
```
Comment (hanzei) [inline labels/mapping.go]: I would like to keep the core set smaller and usable on every repo. Right now we don't use the issue labels and HW labels on every repo (and I think we won't do this in the future). I would like to only have `pullRequest` here for now.

File: labels/mapping.go (hunk 9 above)
Comment (levb) [reply labels/mapping.go:62]: For a newbie (first ticket) - ReactJS may not be most intention-revealing. If anything, I'd then 0/5 prefer to drop `Tech/ReactJS` in favor of `Tech/JavaScript`. Also, having both is more consistent with `mattermost-server`

File: labels/main.go (hunk 8 above)
Comment (levb) [reply labels/main.go]: Made all options explicit.

File: labels/mapping.go (hunk 10 above)
Comment (levb) [reply labels/mapping.go]: Done, also removed HW from workflow since it's closed-source.

**Review by hanzei — APPROVED** [review]

**Comment by hanzei** [comment]
Sorry for taking to long to review this.

**Pull request merged**